L80-csharp-load-balancing.md

```diff
@@ -0,0 +1,259 @@
+        protected virtual void Dispose(bool disposing);
+    }
+    public abstract class LoadBalancerFactory {
+        protected LoadBalancerFactory();
+        public abstract string Name { get; }
+        public abstract LoadBalancer Create(IConnectionController controller, IDictionary<string, object> options);
+    }
+    public class PickContext {
+        public PickContext();
+        public HttpRequestMessage? Request { get; set; }
+        public bool WaitForReady { get; set; }
+    }
+    public class PickFirstBalancer : LoadBalancer {
+        public PickFirstBalancer(IConnectionController channel, ILoggerFactory loggerFactory);
+        public override void ResolverError(Exception exception);
+        public override void UpdateChannelState(ChannelState state);
+        public override void UpdateSubChannelState(SubChannel subChannel, SubChannelState state);
+    }
+    public class PickFirstBalancerFactory : LoadBalancerFactory {
+        public PickFirstBalancerFactory(ILoggerFactory loggerFactory);
+        public override string Name { get; }
+        public override LoadBalancer Create(IConnectionController controller, IDictionary<string, object> options);
+    }
+    public class PickResult {
+        public PickResult(SubChannel? subChannel, DnsEndPoint? endPoint, Action<CompleteContext>? onComplete);
+        public DnsEndPoint? EndPoint { get; }
+        public SubChannel? SubChannel { get; }
+        public void Complete(CompleteContext context);
+    }
+    public class RoundRobinBalancer : SubChannelsLoadBalancer {
+        public RoundRobinBalancer(IConnectionController channel, ILoggerFactory loggerFactory);
+        protected override SubChannelPicker CreatePicker(List<(DnsEndPoint Address, SubChannel SubChannel)> readySubChannels);
+    }
+    public class RoundRobinBalancerFactory : LoadBalancerFactory {
+        public RoundRobinBalancerFactory(ILoggerFactory loggerFactory);
+        public override string Name { get; }
+        public override LoadBalancer Create(IConnectionController controller, IDictionary<string, object> options);
+    }
+    public class StaticAddressResolver : AddressResolver {
+        public StaticAddressResolver(IEnumerable<DnsEndPoint> addresses);
+        public override Task RefreshAsync(CancellationToken cancellationToken);
+        public override IDisposable Subscribe(IObserver<AddressResolverResult> observer);
+        protected override void Dispose(bool disposing);
+    }
+    public class StaticAddressResolverFactory : AddressResolverFactory {
+        public StaticAddressResolverFactory(IEnumerable<DnsEndPoint> addresses);
+        public override string Name { get; }
+        public override AddressResolver Create(Uri address, AddressResolverOptions options);
+    }
+    public class SubChannel : IDisposable {
+        public BalancerAttributes Attributes { get; }
+        public DnsEndPoint? CurrentEndPoint { get; }
+        public ConnectivityState State { get; }
```

Comment:
This class also needs a way for the LB policy to subscribe to connectivity state updates on an ongoing basis.

Reply:
That update is notified by a method call on `LoadBalancer`: `void UpdateSubChannelState(SubChannel subChannel, SubChannelState state);`

Reply:
Resolving this in favor of the thread above.